### DISCUSSION TITLE
Custom admin interface title and favicon

### DISCUSSION THREAD

**Discussion by ali-d-coded** [keystonejs/keystone#7506] [Feature requests]
How can I remove keystone's default title and favicon and replace it with my own? is it possible?

**Comment by raveling** [comment]
Great idea @Ali-D-Coded 

This isn't currently possible. I'm making this a feature request 🙇

**Comment by matchboxhero** [comment]
Did this get implemented?

**Comment by DiesIrae** [comment]
For title, I found a simple hack:
- Implement custom navigation
- Add this hook to it, with the new title base you want:

```typescript
import { useEffect } from "react"

export const useCustomTitle = (cmsTitleBase: string) =>
  useEffect(() => {
    const currentTitle = document.title
    const newTitle = currentTitle.replace("Keystone", cmsTitleBase)
    if (newTitle !== currentTitle) document.title = newTitle
  })

```

**Reply by jj-matos** [reply]
Couldn't make this work. Where was it implemented? The custom navigation is not loaded at sign in page. Tried in layout component, but also only loads after sign in.

**Comment by saleehk** [comment]
You can replace existing favicon with the following method

```js
getAdditionalFiles: [
        async (config: KeystoneConfig) => [
          {
            mode: 'copy',
            inputPath: Path.join(pkgDir, 'public', 'favicon.ico'),
            outputPath: 'public/favicon.ico',
          },
        ],
      ],
```

**Reply by oldo** [reply]
Thanks for this solution, but where do I set `getAdditionalFiles`?

**Reply by MohammadKurjieh** [reply]
@oldo this is added to the `ui` field of the keystone config.
Docs: [Github](https://github.com/keystonejs/keystone/blob/3d20b94f8791225bf832b4c8d83d9235b6cc103b/docs/pages/docs/config/config.md?plain=1#L164) or  [Keystone website](https://keystonejs.com/docs/config/config#ui)

**Reply by oldo** [reply]
Haha, fancy seeing you here @MohammadKurjieh 
Thanks!

**Reply by jmchor** [reply]
For the full picture here:

```typescript
import path from 'path'; 
const pkgDir = path.resolve(__dirname, '..');
```
an in the `ui` field of `keystone.ts`:
```typescript
getAdditionalFiles: [
        async (config: KeystoneConfig) => [ //it seemed as though you could omit the config here though
          {
            mode: 'copy',
            inputPath: Path.join(pkgDir, 'public', 'favicon.ico'),
            outputPath: 'public/favicon.ico',
          },
        ],
      ],
```

You need to make sure that you have a "favicon.ico" file in your `public` folder.

**Comment by molotow11** [comment]
`useCustomTitle` method for change <title> seems not completely works for me. Only on page reload and not on the login page.

**Reply by DiesIrae** [reply]
yeah in practise it seems not to work on all cases. I found that it doesn't always work after logging. Maybe by adding a `setTimeout` or `useLayoutEffect`?

**Reply by molotow11** [reply]
No, this works the same for me with `useLayoutEffect` or `setTimeout`, I need reload a page after login for see title change.

**Comment by jj-matos** [comment]
I managed to do it by putting the new favicon.ico in the public folder, and then adding the public folder as a storage space in keystone.ts. In the end it worked just like the logo, which also only works per documentation if we put the file in a storage space existent in keystone.ts:

Apart from the Logo settings in config.tsx per documentation:
- The files need to be copied to public folder;
- Local sotrage needs to be created in keystone.ts:

```
      app_logos: {
        kind: 'local',
        type: 'image',
        generateUrl: path => `${baseurl}/${path}`,
        serverRoute: {
          path: '/',
        },
        storagePath: 'public',
      },
```

**Comment by MauriceArikoglu** [comment]
Is this supported?
